### PR TITLE
fix(front): set <Form /> (rjsf) as a controlled component

### PR DESCRIPTION
**Pull request by asf-algoo**
https://github.com/tracim/tracim/issues/4105

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

### CODE REVIEW

File: functionnal_tests/cypress/integration/profile/edit_react_json_schema_form.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            url: `/api/workspaces/${baseWorkspace.workspace_id}/contents`,
+            auth: {
+              user: baseOtherUser.email,
+              pass: baseOtherUser.password
+            },
+            body: {
+              content_type: 'html-document',
+              label: 'some note'
+            }
+          }).then(() => {
```
Comment (grignards) [inline functionnal_tests/cypress/integration/profile/edit_react_json_schema_form.js:43]: I think the `then()` is not needed here as `cy.` commands are always executed sequentially, aren't they?

**Review by grignards — APPROVED** [review]

File: functionnal_tests/cypress/integration/profile/edit_react_json_schema_form.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    cy.resetDB()
+    cy.setupBaseDB()
+    cy.login(defaultAdmin)
+    cy.visitPage({ pageName: PAGES.PROFILE, params: { userId: defaultAdmin.user_id } })
+  })
+
+  describe('Information form', () => {
+    describe('while receiving a TLM', () => {
+      it('should not have reset the value of the first input', () => {
+        cy.get(`[data-cy=CustomFormManager__updateProfile__edit__button]`)
```
Comment (gbonaspetti) [inline functionnal_tests/cypress/integration/profile/edit_react_json_schema_form.js:17]:
```suggestion
        cy.get('[data-cy=CustomFormManager__updateProfile__edit__button]')
```

File: frontend/src/component/PublicProfile/CustomFormManager.jsx [hunk 3]
```diff
@@ -206,28 +207,46 @@ const TextRichWidget = connect(({ user }) => ({ user }))(props => {
...
-    >
-      <IconButton
-        customClass={props.submitButtonClass}
-        icon='check'
-        type='submit'
-        text={props.validateLabel}
-      />
-    </Form>
-  )
+class SchemaAsForm extends React.Component {
```
Comment (gbonaspetti) [inline frontend/src/component/PublicProfile/CustomFormManager.jsx:210]: Shouldn't this be in a different jsx file ?

**Review by gbonaspetti — CHANGES_REQUESTED** [review]

File: functionnal_tests/cypress/integration/profile/edit_react_json_schema_form.js (hunk 1 above)
Comment (Skylsmoi) [reply functionnal_tests/cypress/integration/profile/edit_react_json_schema_form.js:43]: I think your right, I will try

File: functionnal_tests/cypress/integration/profile/edit_react_json_schema_form.js (hunk 2 above)
Comment (Skylsmoi) [reply functionnal_tests/cypress/integration/profile/edit_react_json_schema_form.js:17]: will fix

File: frontend/src/component/PublicProfile/CustomFormManager.jsx (hunk 3 above)
Comment (Skylsmoi) [reply frontend/src/component/PublicProfile/CustomFormManager.jsx:210]: this file holds 6 components
each of them could be in a specific file but I chose not to to save development time

once we will refactor it to make it more generic with the app custom forms, we will can split it

File: functionnal_tests/cypress/integration/profile/edit_react_json_schema_form.js (hunk 1 above)
Comment (Skylsmoi) [reply functionnal_tests/cypress/integration/profile/edit_react_json_schema_form.js:43]: without the .then, the test sometimes fails randomly in the `expect()` by testing `'s' === 'some text'` so I'm assuming it's important.

Won't fix

File: frontend/src/component/PublicProfile/CustomFormManager.jsx (hunk 3 above)
Comment (gbonaspetti) [reply frontend/src/component/PublicProfile/CustomFormManager.jsx:210]: ok, works for me

**Review by gbonaspetti — APPROVED** [review]

**Pull request merged**